.github/workflows/ci.yml

```diff
@@ -0,0 +1,57 @@
+name: CI
+
+on:
+  pull_request:
+
+# Python Versions Reference: https://github.com/home-assistant/architecture/blob/master/adr/0002-minimum-supported-python-version.md
+
+jobs:
+  test:
+    strategy:
+      matrix:
+        python-version: [3.11, 3.12]
+    name: Run tests for ${{ matrix.python-version }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - uses: actions/setup-python@v5
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - uses: snok/install-poetry@v1
+
+      - name: Install Dependencies
+        run: poetry install
+
+      - name: Run tests
+        run: poetry run pytest
+
+  lint:
+    strategy:
+      matrix:
+        python-version: [3.11, 3.12]
```

Comment:
Do we need multiple Python versions for linting?

Reply:
Good question, I was not sure how Python works (there might be differences per version) so I rather just put both. To which one should I switch, the latest? 🙂

Reply:
If we perform linting with multiple Python versions and encounter differences, addressing those issues without ignoring rules could be challenging. And Home Assistant now supports Python 3.12, so I suggest that we should target that.